### DISCUSSION TITLE
Bringing own text "renderer"/preprocessor

### DISCUSSION THREAD

**Discussion by derdilla** [freeplane/freeplane#2132] [Help]
I recently started using [typst](https://github.com/typst/typst/) (tldr: its a binary to turn text to svg/png/pdf, like latex) regularly and wanted to use it to add some formulas to my graphs.
Since freeplane has an xml html-ish format I thought I might give implementing a add-on/script to:
1. Parse all text
2. Replace them with svgs during "rendering"(if there is such a thing)

Before I needlessly sink my hours in there I would like to ask:
- How feasible is this concept in freeplane?
- Does anyone have pointers on where to start? (script or add-on or even fork, apis to look into)

**Comment by euu2021** [comment]
I remember those threads, that can be helpful:
https://github.com/freeplane/freeplane/discussions/568
https://github.com/freeplane/freeplane/discussions/1640
https://github.com/freeplane/freeplane/discussions/1115

Also, looking at how the Insert Inline Image add-on works, may be useful:
https://docs.freeplane.org/getting-started/Add-ons_(install).html#insert-inline-image
